messaging/src/main/java/org/axonframework/eventhandling/deadletter/jpa/JpaSequencedDeadLetterQueue.java

```diff
@@ -0,0 +1,716 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.eventhandling.deadletter.jpa;
+
+import com.thoughtworks.xstream.XStream;
+import org.axonframework.common.AxonConfigurationException;
+import org.axonframework.common.jpa.EntityManagerProvider;
+import org.axonframework.common.transaction.TransactionManager;
+import org.axonframework.eventhandling.EventMessage;
+import org.axonframework.messaging.Message;
+import org.axonframework.messaging.MetaData;
+import org.axonframework.messaging.deadletter.Cause;
+import org.axonframework.messaging.deadletter.DeadLetter;
+import org.axonframework.messaging.deadletter.DeadLetterQueueOverflowException;
+import org.axonframework.messaging.deadletter.EnqueueDecision;
+import org.axonframework.messaging.deadletter.GenericDeadLetter;
+import org.axonframework.messaging.deadletter.NoSuchDeadLetterException;
+import org.axonframework.messaging.deadletter.SequencedDeadLetterQueue;
+import org.axonframework.serialization.Serializer;
+import org.axonframework.serialization.SimpleSerializedObject;
+import org.axonframework.serialization.xml.CompactDriver;
+import org.axonframework.serialization.xml.XStreamSerializer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.invoke.MethodHandles;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import javax.annotation.Nonnull;
+import javax.persistence.EntityManager;
+import javax.persistence.NoResultException;
+
+import static org.axonframework.common.BuilderUtils.assertNonNull;
+import static org.axonframework.common.BuilderUtils.assertThat;
+
+/**
+ * JPA-backed implementation of the {@link SequencedDeadLetterQueue}, used for storing dead letters containing
+ * {@link EventMessage Eventmessages} as a {@link DeadLetterEntry}.
+ * <p>
+ * Keeps the insertion order intact by saving an incremented index within each sequence, backed by the
+ * {@link DeadLetterEntry#getIndex()} property.
+ * <p>
+ * When processing an item, single execution across all applications is guaranteerd by setting the
+ * {@link DeadLetterEntry#getProcessingStarted()} property, locking other processes out of the sequence for the
+ * configured {@code claimDuration} (30 seconds by default).
+ * <p>
+ * Configuring a chain of upcasters is not supported.
+ *
+ * @param <M> An implementation of {@link Message} contained in the {@link DeadLetter dead-letters} within this queue.
+ * @author Mitchell Herrijgers
+ * @since 4.6.0
+ */
+public class JpaSequencedDeadLetterQueue<M extends EventMessage<?>> implements SequencedDeadLetterQueue<M> {
+
+    private static final Logger logger = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
```

Comment:
I would expect this `logger` to be used a bit more.
That way, if we do get requests from users experiencing issues with the `JpaSequencedDeadLetterQueue`, we can ask them to provides the logs too.
Differently put, it'll help us in the future for sure.